### DISCUSSION TITLE
domain rules minimum TTL doesn't work

### DISCUSSION THREAD

**Discussion by c-michail** [mokeyish/smartdns-rs#373] [General]
Hello,
  I'm setting "rr-ttl-min 600"  in the config and i use domain-rules to exclude a domain from 600 secs TTL but it doesn't work, 
  it always answers with 600 TTL.

  Domain rule:
  domain-rules /www.mydomain.com/ -no-cache -no-serve-expired -rr-ttl-min 60

Any ideas ?

Thank you

**Comment by c-michail** [comment]
2024-09-16 17:53:57.514:WARN:smartdns::config::parser::domain_rule:54: domain rule: unkown options no-serve-expired=None
2024-09-16 17:53:57.514:WARN:smartdns::config::parser::domain_rule:54: domain rule: unkown options rr-ttl-min=Some("60")
2024-09-16 17:53:57.514:WARN:smartdns::config::parser::domain_rule:54: domain rule: unkown options no-serve-expired=None
2024-09-16 17:53:57.514:WARN:smartdns::config::parser::domain_rule:54: domain rule: unkown options rr-ttl-min=Some("60")

**Reply by mokeyish** [reply]
It seems there is a bug, I will fix it next time.

**Reply by c-michail** [reply]
@mokeyish not fixed in v0.8.8-alpha.1

**Reply by mokeyish** [reply]
I forgot, I'll deal with it next time

**Reply by mokeyish** [reply]
This should be fixed by #408 , and you can download from here https://github.com/mokeyish/smartdns-rs/actions/runs/11274524312

**Reply by c-michail** [reply]
Hello,
  You forgot 2 more options:
  ![εικόνα](https://github.com/user-attachments/assets/34973935-3900-4540-9685-0164edd5e576)

**Reply by c-michail** [reply]
Also it doesn't seem to work/override. I have in my config: 

rr-ttl-min 1800
domain-rules /mydomain.com/ -no-cache -rr-ttl-min 60

Smartdns will still return 1800 TTL of "mydomain.com" (like it ignores the rule above)

**Reply by c-michail** [reply]
@mokeyish did you also check the above ?

**Comment by mokeyish** [comment]
There are many things that have not been implmented😅. I have limited energy, so I can only do it slowly. It would be great if you could also participate.

**Reply by c-michail** [reply]
Sure i understand. I tried to add the Domain prefetch interval as config parameter and i messed it up 😄 (i am not a rust dev still learning)

**Reply by mokeyish** [reply]
Learning Rust is quite strange. It has different difficulty levels for people from different backgrounds. Some people find it easy, while others find it complicated. But you can look at my PRs. Sometimes, adding a feature actually only takes a small amount of code.

**Comment by mokeyish** [comment]
#410

**Reply by c-michail** [reply]
Hello,
  Can you add no caching and ignore rr-ttl too ?
  I mean to ignore those for the specific domain.

Thank you
